src/System.Numerics.Vectors/src/System.Numerics.Vectors.csproj

```diff
@@ -46,16 +46,32 @@
   </ItemGroup>
   <!-- Carry a copy of MathF where not available -->
   <ItemGroup Condition="'$(IsPartialFacadeAssembly)' == 'true' OR $(TargetGroup.StartsWith('netstandard'))">
-    <Compile Include="..\..\Common\src\System\MathF.netstandard.cs">
+    <Compile Include="$(CommonPath)\System\MathF.netstandard.cs">
       <Link>System\MathF.netstandard.cs</Link>
     </Compile>
   </ItemGroup>
+  <!-- Optimize Quaternion as Vector4 for netcoreapp -->
+  <!-- Jit issue for other runtimes https://github.com/dotnet/coreclr/issues/15237 -->
+  <ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
+    <Compile Include="System\Numerics\Quaternion.netcoreapp.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\Internal\Runtime\CompilerServices\Unsafe.cs">
+        <Link>Common\CoreLib\Internal\Runtime\CompilerServices\Unsafe.cs</Link>
```

Comment:
You need to reference the internal Unsafe in CoreLib. Local copy is not going to work - it won't be recognized by the JIT.

Reply:
Like this?
```xml
<ItemGroup Condition="'$(TargetGroup)' == 'netcoreapp'">
 <Compile Include="System\Numerics\Quaternion.netcoreapp.cs" />
 <ReferenceFromRuntime Include="System.Private.CoreLib" />
</ItemGroup>
<ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' AND '$(TargetGroup)' != 'netcoreapp'">
 <Compile Include="System\Numerics\Quaternion.cs" />
</ItemGroup>
```

Reply:
Yes, something like this.